runtime/lib/component/custodian.js

```diff
@@ -1,239 +1,268 @@
-       * although there is no WiFi history, yet some app is running out there,
-       * continuing currently app.
-       */
-      logger.info('no WiFi history exists, continuing currently running app.')
-      return this.component.light.ttsSound('@yoda', 'system://guide_config_network.ogg')
-        .then(() =>
-        /** awaken is not set for no network available, recover media directly */
-          this.component.turen.recoverPausedOnAwaken()
-        )
+  clearTimeout(this.bleTimer)
+  this.bleTimer = setTimeout(() => {
+    logger.debug(`closeBluetooth, runtimeState: ${this.runtimeState}`)
+    logger.debug(`closeBluetooth, preRuntimeState: ${this.preRuntimeState}`)
+
+    if (this.runtimeState === RuntimeState.CONFIGURING_NETWORK) {
+      this.runtimeState = this.preRuntimeState
+    }
+
+    this.bleOpened = false
+    this.component.light.stop('@yoda', 'system://setStandby.js')
+    if (delaySecond < 2000) {
+      setTimeout(() => this.bluetoothStream.end(), 2000)
+    } else {
+      this.bluetoothStream.end()
     }
-    /**
-     * No WiFi connection history found, introduce device setup procedure.
-     */
-    logger.info('no WiFi history exists, announcing guide to network configuration.')
-    return this.component.light.ttsSound('@yoda', 'system://guide_config_network.ogg')
-      .then(() =>
-        /** awaken is not set for no network available, recover media directly */
-        this.component.turen.recoverPausedOnAwaken()
-      )
+  }, delaySecond)
+}
+
+Custodian.prototype.turenDidWakeUp = function () {
+  logger.debug(`turenDisWakeUp, runtimeState: ${this.runtimeState}`)
+
+  if (this.runtimeState === RuntimeState.LOGGED_IN) {
+    return
   }
+  this.component.turen.pickup(false)
 
-  /**
-   * if runtime is logging in or network is unavailable,
-   * and there is WiFi history existing,
-   * announce WiFi is connecting.
-   */
-  logger.info('announcing network connecting on voice coming.')
-  wifi.enableScanPassively()
-  return this.component.light.ttsSound('@yoda', 'system://wifi_is_connecting.ogg')
-    .then(() =>
-      /** awaken is not set for no network available, recover media directly */
-      this.component.turen.recoverPausedOnAwaken()
-    )
-}
-
-// MARK: - END Interception
+  if (this.runtimeState === RuntimeState.RAW_NETWORK ||
+      this.runtimeState === RuntimeState.CONFIGURING_NETWORK) {
+    /* awaken is not set for no network available, recover media directly */
+    return this.component.light.ttsSound(
+      '@yoda', 'system://guide_config_network.ogg'
+    ).then(() => this.component.turen.recoverPausedOnAwaken())
+  } else if (this.runtimeState === RuntimeState.CONNECTING_NETWORK ||
+             this.runtimeState === RuntimeState.LOGGING) {
+    /* awaken is not set for no network available, recover media directly */
+    return this.component.light.ttsSound(
+      '@yoda', 'system://wifi_is_connecting.ogg'
+    ).then(() => this.component.turen.recoverPausedOnAwaken())
+  }
+}
```

Comment:
You have changed source of custodian, so you need to fix and add test cases, too.